docs/how-tos/typescript_consume.md

```diff
@@ -0,0 +1,161 @@
+
+- `react-relay` for React integration with Relay
+- `relay-runtime` for core Relay functionality
+- `babel-plugin-relay` for compiling Relay queries
+
+## Setup Relay Environment
+
+We initially have to setup a Relay environment that helps the Relay to access the GraphQL endpoint.
+In the root of `src` we create a `RelayEnvironment.ts`. In this file we are setup up a Relay environment for making GraphQL requests.
+The fetchFn function sends a POST request to the GraphQL endpoint with the query and variables, and processes the response.
+The `createRelayEnvironment` function initializes the Relay environment by combining the network layer (responsible for fetching data) and the store (which caches the data).
+
+!!! example "RelayEnvironment.ts"
+
+    ```typescript
+    import {
+        Environment,
+        Network,
+        RecordSource,
+        Store,
+        FetchFunction,
+    } from "relay-runtime";
+
+    const HTTP_ENDPOINT = "http://127.0.0.1:8000/graphql"; // assuming you are running any of the services from how to guides
+
+    const fetchFn: FetchFunction = async (request, variables) => {
+        const response = await fetch(HTTP_ENDPOINT, {
+            method: "POST",
+            headers: {
+                "Content-Type": "application/json",
+            },
+            body: JSON.stringify({
+                query: request.text,
+                variables,
+            }),
+        });
+
+        return await response.json();
+    };
+
+    function createRelayEnvironment() {
+        return new Environment({
+        network: Network.create(fetchFn),
+        store: new Store(new RecordSource()),
+        });
+    }
+
+    export const RelayEnvironment = createRelayEnvironment();
+
+    ```
+
+!!! info
+
+    Refer to the [Relay Environment Docs](https://relay.dev/docs/v1.6.2/relay-environment/) to learn more.
+
+## Generate TypeScript Types
+
+Relay generated TypeScript types automatically for your queries, based on the GraphQL schema.
+For examples, if you are querying a Person type form the GraphQL schema, Relay will generate types for you.
+
+!!! example "GraphQL Query"
+
+    ```graphql
+    query PersonQuery {
+        person {
+            id
+            firstName
+            lastName
+        }
+    }
+    ```
+
+Relay will automatically generate a TypeScript type for this Query when you run the Relay compiler.
+
+!!! example "Generated Type"
+
+    ```typescript
+    // __generated__/PersonQuery.graphql.ts
+
+    export interface PersonQuery {
+        person: {
+            id: string;
+            firstName: string;
+            lastName: string;
+        } | null;
+    }
+    ```
+
+## Fetch Data Using Relay
+
+In Relay, you defing your GraphQL queries within components using the `useLazyLoadQuery` or `useFragment` hooks. Relay then fetched and managed data for you.
```

Comment:
As Benedikt [recently pointed out](https://github.com/DiamondLightSource/graph-federation/pull/48#pullrequestreview-2340825957), it's nice to stick to we / our instead of switching to you / your in places. Also good to keep things in the future tense if that's what we're going with throughout

```suggestion
In Relay, we define our GraphQL queries within components using the `useLazyLoadQuery` or `useFragment` hooks. Relay then fetches and manages data for you.
```